### DISCUSSION TITLE
Connection with a NAS Server

### DISCUSSION THREAD

**Discussion by marcosfa00** [community/community#102739] [Programming Help]
Hello guys, I want to have a connection with my **NAS** server in where I have my website, but the problem is that my router does not allow me to connect this server by lan with the domain address, do you know any solution to this?

**Comment by My-Odyssey** [comment]
1. Set up a local DNS server -> Install and configure a local DNS server (Pi-hole, AdGuard) on your network.
2. Configure a DNS resolution -> Add a DNS record to the local DNS server to resolve your domain to the NAS's internal IP address.
3. Distribute DNS server IP -> Configure your router or DHCP server to distribute the local DNS server IP address to all devices on your network.

Pi-hole or other selfhosted DNS server are usually light weight and easy to install. Just let it run in a Docker on your NAS or on a Raspberry.

When outside the network, you need to either connect to the network directly or configure port forwarding on your router. The specific port you need to forward depends on your use case. Alternatively, you can use a VPN to connect to your home network. However, you will need to use the local IP address when connected via VPN.

If you are within the local network, there's usually no need to use the domain name to access your website, using the local IP address is simpler and achieves the same result.
